Runtime/Scripts/BaseMeshSync.Textures.cs

```diff
@@ -0,0 +1,210 @@
+            var savePath = GetSavePath(mat, textureName);
+
+            // LoadAssetAtPath can throw an exception if it's called during a domain backup but we have no way to detect that:
+            try {
+                if (AssetDatabase.LoadAssetAtPath<Texture2D>(savePath) == null) {
+                    EditorApplication.delayCall -= UpdatePendingMaterials;
+                    EditorApplication.delayCall += UpdatePendingMaterials;
+
+                    return false;
+                }
+            }
+            catch (Exception ex) {
+                return false;
+            }
+
+            // Ensure the texture is up to date:
+            AssetDatabase.ImportAsset(savePath);
+
+            var savedTexture = AssetDatabase.LoadAssetAtPath<Texture2D>(savePath);
+
+            mat.SetTexture(textureName, savedTexture);
+
+            if (textureName == MeshSyncConstants._MetallicGlossMap ||
+                textureName == MeshSyncConstants._MaskMap) {
+                mat.EnableKeyword(MeshSyncConstants._METALLICGLOSSMAP);
+                mat.EnableKeyword(MeshSyncConstants._METALLICSPECGLOSSMAP);
+
+                TextureImporter importer = (TextureImporter)AssetImporter.GetAtPath(savePath);
+                if (importer != null) {
+                    importer.sRGBTexture = false;
+                    importer.alphaIsTransparency = false;
+                    importer.ignorePngGamma = true;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to apply any pending material updates and unschedules itself when done.
+        /// </summary>
+        private void UpdatePendingMaterials() {
+            EditorApplication.delayCall -= UpdatePendingMaterials;
+
+            for (int i = pendingMaterialUpdates.Count - 1; i >= 0; i--) {
+                Tuple<Material, string> pendingMaterial = pendingMaterialUpdates[i];
+                if (SetSerializedTextureForMaterial(pendingMaterial.Item1, pendingMaterial.Item2)) {
+                    pendingMaterialUpdates.RemoveAt(i);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Sets the required shader for the given material.
+        /// </summary>
+        /// <param name="mat">Material to set the shader on</param>
+        /// <param name="shaderName">The name of the shader passed by the DCC tool</param>
+        private static void UpdateShader(ref Material mat, string shaderName) {
+            Shader shader = null;
+            if (!string.IsNullOrEmpty(shaderName)) {
+                shader = Shader.Find(shaderName);
+            }
+
+            bool shaderExists = shader != null;
+
+            if (shader == null) {
+#if AT_USE_HDRP
+                shader = Shader.Find("HDRP/Lit");
+#elif AT_USE_URP
+                shader = Shader.Find("Universal Render Pipeline/Lit");
+#else
+                shader = Shader.Find("Standard");
+#endif
```

Comment:
Could there be a property that fetches the Lit shader name, to encapsulate this logic?

Reply:
Done